### PR TITLE
Refactor/add victim

### PR DESCRIPTION
**Pull request by themanforfree**
Please briefly answer these questions:
described in #253 
based on #262 
* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
handle response channel block
* what changes does this pull request make?
add `sync_victims_task` to retry blocked response
* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
no

### CODE REVIEW

File: xline/src/storage/kvwatcher.rs [hunk 1]
```diff
@@ -125,62 +170,103 @@ where
...
+                .insert(watch_id),
             "can't insert a watcher twice"
         );
     }
 
+    /// Move a watcher to victims, the `watch_id` must be valid.
+    fn move_to_victim(&mut self, watch_id: WatchId, updates: (i64, Vec<Event>)) {
+        debug!(watch_id, "move watcher to victim");
+        let Some(watcher) = self.watchers.remove(&watch_id) else {
+            panic!("watcher should exist")
```
Comment (rogercloud) [inline xline/src/storage/kvwatcher.rs:218]: unreachable instead?

Comment (themanforfree) [reply xline/src/storage/kvwatcher.rs]: Fixed

File: xline/src/storage/kvwatcher.rs [hunk 2]
```diff
@@ -428,4 +597,50 @@ mod test {
         }
         handle.abort();
     }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 10)]
+    async fn test_victim() {
+        let (store, db, kv_watcher) = init_empty_store();
+        // response channel with capacity 1, so it will be full easily, than we can trigger victim
```
[the remark this answers is not in the document]
Comment (themanforfree) [reply xline/src/storage/kvwatcher.rs]: Fixed

File: xline/src/storage/kvwatcher.rs [hunk 3]
```diff
@@ -262,25 +359,81 @@ where
...
+                while let Some(updates) = kv_update_rx.recv().await {
+                    kv_watcher.handle_kv_updates(updates);
+                }
             }
         });
         kv_watcher
     }
 
+    /// Background task to sync victims
+    async fn sync_victims_task(self: Arc<Self>) {
```
Comment (Phoenix500526) [inline xline/src/storage/kvwatcher.rs:380]: Shall we put this background task under the control of shutdown_trigger like some other background tasks does?

**Review by Phoenix500526 — COMMENTED** [review]
I recommend enlarging the default channel size from 128 to 1024. 128 looks a little too small.

File: xline/src/storage/kvwatcher.rs [hunk 4]
```diff
@@ -262,25 +359,81 @@ where
...
+                        watche_id = watcher.watch_id(),
+                        "watcher synced by sync_victims_task"
+                    );
+                    watcher_map_w.register(watcher);
+                }
+            }
+            if !new_victims.is_empty() {
+                self.watcher_map.write().victims.extend(new_victims);
+            }
+            sleep(Duration::from_millis(10)).await;
```
Comment (rogercloud) [inline xline/src/storage/kvwatcher.rs:427]: make `10` a const value or a configurable value.

File: xline/src/storage/kvwatcher.rs [hunk 5]
```diff
@@ -424,4 +597,50 @@ mod test {
         }
         handle.abort();
     }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 10)]
+    async fn test_victim() {
```
Comment (rogercloud) [inline xline/src/storage/kvwatcher.rs:602]: Is there a test to cover watch cancel logic?

File: xline/src/storage/kvwatcher.rs (hunk 4 above)
Comment (themanforfree) [reply xline/src/storage/kvwatcher.rs]: Added

File: xline/src/storage/kvwatcher.rs (hunk 5 above)
Comment (themanforfree) [reply xline/src/storage/kvwatcher.rs:619]: Added

**Review by rogercloud — DISMISSED** [review]

File: utils/src/config.rs [hunk 6]
```diff
@@ -367,6 +374,41 @@ impl Default for ClientTimeout {
     }
 }
 
+/// Curp client settings
```
Comment (Phoenix500526) [inline utils/src/config.rs:377]: Plz modify this comment

**Review by Phoenix500526 — APPROVED** [review]

**Review by rogercloud — APPROVED** [review]

**Review by rogercloud — APPROVED** [review]

**Pull request merged**